front/tests/008-train-schedule.spec.ts

```diff
@@ -0,0 +1,109 @@
+});
+
+test.beforeEach(async ({ page }) => {
+  const homePage = new HomePage(page);
+  await homePage.goToHomePage();
+  selectedLanguage = await homePage.getOSRDLanguage();
+
+  // Navigate to the created scenario page
+  await page.goto(
+    `/operational-studies/projects/${project.id}/studies/${study.id}/scenarios/${scenario.id}`
+  );
+});
+
+test.describe('Verifying that all elements in the train schedule are loaded correctly', () => {
+  test('should accurately load the trains with their corresponding simulation results', async ({
+    page,
+  }) => {
+    test.setTimeout(120000);
+    const scenarioPage = new ScenarioPage(page);
+    const opTimetablePage = new OperationalStudiesTimetablePage(page);
+
+    // Check if the infrastructure is loaded
+    await scenarioPage.checkInfraLoaded();
+
+    // Verify the train count and various elements on the timetable page
+    await opTimetablePage.verifyTrainCount(20, selectedLanguage);
+    await opTimetablePage.verifyInvalidTrainsMessageVisibility(selectedLanguage);
+    await opTimetablePage.checkSelectedTimetableTrain();
+    await opTimetablePage.filterValidityAndVerifyTrainCount(selectedLanguage, 'Valid', 16);
+    await opTimetablePage.verifyEachTrainSimulation();
+  });
+
+  test('should accurately filter the imported trains', async ({ page }) => {
+    test.setTimeout(120000);
+    const scenarioPage = new ScenarioPage(page);
+    const opTimetablePage = new OperationalStudiesTimetablePage(page);
+
+    // Check if the infrastructure is loaded
+    await scenarioPage.checkInfraLoaded();
+
+    // Verify the train count and apply various filters
+    await opTimetablePage.verifyTrainCount(20, selectedLanguage);
+    await opTimetablePage.filterValidityAndVerifyTrainCount(selectedLanguage, 'Invalid', 4);
+    await opTimetablePage.filterValidityAndVerifyTrainCount(selectedLanguage, 'All', 20);
+    await opTimetablePage.filterHonoredAndVerifyTrainCount(selectedLanguage, 'Honored', 12);
+    await opTimetablePage.filterValidityAndVerifyTrainCount(selectedLanguage, 'Valid', 12);
+    await opTimetablePage.filterHonoredAndVerifyTrainCount(selectedLanguage, 'Not honored', 4);
+    await opTimetablePage.filterValidityAndVerifyTrainCount(selectedLanguage, 'Invalid', 0);
+    await opTimetablePage.filterHonoredAndVerifyTrainCount(selectedLanguage, 'All', 4);
+    await opTimetablePage.filterValidityAndVerifyTrainCount(selectedLanguage, 'All', 20);
+
+    // Define the composition filters and verify each filter
+    const compositionFilters = [
+      { code: 'MA100', count: 7 },
+      { code: 'HLP', count: 3 },
+      { code: 'E32C', count: 1 },
+      { code: 'Without code', count: 9 },
+    ];
+
+    await compositionFilters.reduce(async (promise, filter) => {
+      await promise;
+      return opTimetablePage.clickCodeCompoTrainFilterButton(
+        selectedLanguage,
+        filter.code,
+        filter.count
+      );
+    }, Promise.resolve());
```

Comment:
Using a reduce here is a bit weird since you are not creating a accumulator. Why not use a for loop ?

Reply:
I was using a `for `loop, but it generated ESLint errors. Following Theo's suggestion to avoid those messages, I switched to using `reduce `instead.

Reply:
Changed back to for loop